server/subscribe.go

```diff
@@ -324,6 +324,34 @@ func (p *Plugin) editChannelSubscription(modifiedSubscription *ChannelSubscripti
 	})
 }
 
+func (p *Plugin) listChannelSubscriptions() (string, error) {
+	subs, err := p.getSubscriptions()
+	if err != nil {
+		return "", err
+	}
+
+	rows := make([]string, len(subs.Channel.ById)+len(subs.Channel.IdByChannelId))
```

Comment:
nit: more idiomatic would be initialize rows to an empty array and `append` as you go?

Reply:
Right, I thought this would be more performant (negligibly if that), but certainly less readable and more error-prone. Will change to `append`.

I'm constantly coming back to "since we are parsing the entire subscriptions blob of the KV store for any operation, is any optimization thereafter making a significant difference?" No matter how big the dataset is, we have to read the whole thing from the db and store it in memory. Thus, little attempts at optimization like this block of code above seem pointless.

Reply:
We are talking about a few extra byte copies (string pointers), append doubles the allocation size every time, so the win here is _really_ negligible until the numbers are really large.